src/large_pages/node_large_page.cc

```diff
@@ -141,9 +147,66 @@ static struct text_region FindNodeTextRegion() {
   }
 
   ifs.close();
+#elif defined(__FreeBSD__)
+  struct text_region nregion;
+  nregion.found_text_region = false;
+
+  std::string exename;
+  {
+    char selfexe[PATH_MAX];
+    size_t count;
+    int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+    const size_t miblen = sizeof(mib) / sizeof(mib[0]);
+    if (sysctl(mib, miblen, selfexe, &count, nullptr, 0) == -1) {
+      return nregion;
+    }
+    exename = std::string(selfexe, count);
+  }
+
+  size_t iprocsz;
+  int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_VMMAP, getpid()};
+  const size_t miblen = sizeof(mib) / sizeof(mib[0]);
+  if (sysctl(mib, miblen, nullptr, &iprocsz, nullptr, 0) == -1) {
+    return nregion;
+  }
+
+  auto alg = std::vector<char>(iprocsz);
+
+  iprocsz = iprocsz * 4 / 3;
+  if (sysctl(mib, miblen, alg.data(), &iprocsz, nullptr, 0) == -1) {
+    return nregion;
+  }
+
+  char* start = alg.data();
+  char* end = start + iprocsz;
+
+  while (start < end) {
+    struct kinfo_vmentry* entry =
+       reinterpret_cast<struct kinfo_vmentry*>(start);
```

Comment:
Is alignment a concern here?

By the way, you can probably fit it on one line if you drop the `struct` keywords.